packages/liveblocks-core/src/room.ts

```diff
@@ -1574,12 +1591,23 @@ export function createRoom<
     effects.send(messages);
   }
 
-  function onMessage(event: IWebSocketMessageEvent) {
+  /**
+   * Handles a raw message received on the WebSocket. Can be either a "pong"
+   * (server heartbeat), or a message from the Liveblocks server.
+   */
+  function handleRawSocketMessage(event: IWebSocketMessageEvent) {
     if (event.data === "pong") {
-      clearTimeout(context.timers.pongTimeout);
-      return;
+      transition({ type: "RECEIVE_PONG" });
+    } else {
+      handleServerMessage(event);
     }
+  }
```

Comment:
Notice that this is presorting on the future state where the handling of the "pong" and handling any other incoming messages are separate parts of the room's client. Handling "pong"s is directly related to maintaining the socket connectivity, whereas handling incoming server messages can always happen, and doesn't need a state transition. Hence the split introduced here. **No behavioral change here yet, though.**